lib/codec.js

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {termCodecs} from './codecs/codecMap';
+import {Base64PadCodec} from './codecs/base64Pad';
+import {CborldError} from './error';
+import {CodecMapCodec} from './codecs/codecMap';
+import {UrlCodec} from './codecs/url';
+import {ContextCodec} from './codecs/context';
+import {SimpleTypeCodec} from './codecs/simpleType';
+import {XsdDateTimeCodec} from './codecs/xsdDateTime';
+
+/**
+ * Gets a list of all term codecs available to use when transforming term
+ * values.
+ *
+ * @returns {Array} - An array of all term codecs.
+ */
+export function getTermCodecs() {
+  const codecs = [];
+  termCodecs.forEach((value, key) => {
+    if(typeof key === 'string') {
+      codecs.push(key);
+    }
+  });
+
+  return codecs;
+}
+
+/**
+ * Gets a compression map for the given set of JSON-LD Context URLs.
+ *
+ * @param {object} [args] - The arguments to the function.
+ * @param {Array} [args.contextUrls] - The JSON-LD Context URLs to use when
```

Comment:
If this is optional, we should set a default in the function params. If not optional, we should remove square brackets. Same for the others.

```suggestion
 * @param {Array} [args.contextUrls] - The JSON-LD Context URLs to use when
```

Reply:
Didn't know what the square bracket jsdoc syntax was for :) -- now I do. Will make a pass through the documentation now making those changes.

Reply:
Fixed in df94876cf12e5c134d4dc3724c080d55b33aef53.